sdk/servicebus/service-bus/src/core/messageSender.ts

```diff
@@ -620,9 +620,11 @@ export class MessageSender extends LinkEntity {
           // rhea throws errors with name `TypeError` but not an instance of `TypeError`, so catch them too
           // Errors in such cases do not have user friendy message or call stack
           // So use `getMessagePropertyTypeMismatchError` to get a better error message
-          error = getMessagePropertyTypeMismatchError(data) || error;
+          const errObj = getMessagePropertyTypeMismatchError(data) || error;
+          throw errObj;
```

Comment:
I would suggest `throw(getMessagePropertyTypeMismatchError(data) || error);` instead

Reply:
Do you suggest to put this under the throw under the if statement?

Reply:
This is what I have - 
` catch (error) {
        if (error instanceof TypeError || error.name === "TypeError") {
          // `RheaMessageUtil.encode` can fail if message properties are of invalid type
          // rhea throws errors with name `TypeError` but not an instance of `TypeError`, so catch them too
          // Errors in such cases do not have user friendy message or call stack
          // So use `getMessagePropertyTypeMismatchError` to get a better error message
          throw getMessagePropertyTypeMismatchError(data) || error;
        }
      }`

Reply:
```
 catch (error) {
        if (error instanceof TypeError || error.name === "TypeError") {
         //Comments
          throw getMessagePropertyTypeMismatchError(data) || error;
        }
      }
```

Reply:
```javascript
if (error instanceof TypeError || error.name === "TypeError") {
          throw getMessagePropertyTypeMismatchError(data) || error;
}
throw error
```

Reply:
fixed